packages/firestore/test/util/equality_matcher.ts

```diff
@@ -46,6 +46,12 @@ function customDeepEqual(left: unknown, right: unknown): boolean {
   /**
    * END: Custom compare logic
    */
+  if (left === 0.0 && right === 0.0) {
```

Comment:
You could probably move this inside the `left === right` case.

```ts
if (left === right) {
  if (left === 0) {
    // ...
    return 1 / left === 1 / right;
  } else {
    return true;
  }
}
```

Also, the comment above about "END: Custom compare logic" I can't say I really understand what it means. It seems like there's custom logic both above and below the comment. Could you strike it or make it clear what it's actually talking about?

Reply:
Done. I was debating doing this earlier anyways.